pkg/globalmanager/downstream.go

```diff
@@ -129,37 +130,79 @@ func (dc *DownstreamController) syncModelWithName(nodeName, modelName, namespace
 
 // syncIncrementalJob syncs the incremental learning jobs
 func (dc *DownstreamController) syncIncrementalJob(eventType watch.EventType, job *sednav1.IncrementalLearningJob) error {
-	// Here only propagate to the nodes with non empty name
+	jobConditions := job.Status.Conditions
+	if len(jobConditions) == 0 {
+		return nil
+	}
 
-	// FIXME(llhuii): only the case that all workers having the same nodeName are support,
-	// will support Spec.NodeSelector and differenect nodeName.
-	nodeName := job.Spec.TrainSpec.Template.Spec.NodeName
-	if len(nodeName) == 0 {
-		return fmt.Errorf("empty node name")
+	dataName := job.Spec.Dataset.Name
+	ds, err := dc.client.Datasets(job.Namespace).Get(context.TODO(), dataName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("dataset(%s/%s) not found", job.Namespace, dataName)
 	}
-	dc.injectSecret(job, job.Spec.CredentialName)
+	dsNodeName := ds.Spec.NodeName
+
+	var trainNodeName string
+	var evalNodeName string
 
-	// Sync the model info to edgenode when the job is created
-	if eventType == watch.Added {
-		models := make(map[string]bool)
-		for _, modelName := range []string{
-			job.Spec.InitialModel.Name,
-			job.Spec.DeploySpec.Model.Name,
-		} {
-			models[modelName] = true
+	ann := job.GetAnnotations()
+	if ann != nil {
+		trainNodeName = ann[AnnotationsKeyPrefix+string(sednav1.ILJobTrain)]
+		evalNodeName = ann[AnnotationsKeyPrefix+string(sednav1.ILJobEval)]
+	}
+
+	if eventType == watch.Deleted {
+		dc.messageLayer.SendResourceObject(dsNodeName, eventType, job)
+		if trainNodeName != "" {
+			dc.messageLayer.SendResourceObject(trainNodeName, eventType, job)
 		}
 
-		for modelName := range models {
-			err := dc.syncModelWithName(nodeName, modelName, job.Namespace)
-			if err != nil {
-				klog.Warningf("Error to sync model %s when sync incremental learning job %s to node %s: %v", modelName, job.Name, nodeName, err)
+		if evalNodeName != "" {
+			dc.messageLayer.SendResourceObject(evalNodeName, eventType, job)
+		}
```

Comment:
need to return function here after `watch.Deleted`

Reply:
ok, fix it